### ISSUE TITLE
Professor Putricide -not counting attempts

### ISSUE THREAD

**Issue by Yoni140** [TrinityCore/TrinityCore#6364]
Hey, in the professor putricide you have only 10 tries to kill him , after that you cant reach to him untill the save reset .
now there is nothing that count the attempts and you can do him as much as you want .

http://www.wowhead.com/npc=36678/professor-putricide#comments
"It should be worth it mentioning that you only get 10 tries to kill Professor Putricide during a raid-save. Should the raid group wipe 10 times and come back for an 11th try, Putricide will have despawned and not be showing up again until after the reset of ICC."

Hash : fb787aee5e   Tdb: 335.11.46+
##

**Comment by Keader** [comment]
What happens with Lich King in this case?
You can go to Lich King without kill Professor or you cant reach LK?

**Comment by Treeston** [comment]
Limited attempts should only apply to heroic mode - see [here](https://totemz.wordpress.com/2010/02/09/limited-attempts-removed-from-icecrown-citadel-normal-mode/), [here](http://wow.allakhazam.com/story.html?story=21513) and I'm sure there's more.

**Comment by Treeston** [comment]
Attempts are being processed properly in current core. However, there seems to be a visual issue causing attempt counts to now show up in the client.

**Comment by Treeston** [comment]
This visual issue has some _very_ strange behavior. I've verified the world state IDs used (`4940` show toggle, `4941` current attempts, `4942` max attempts) in 3.3.5 era sniffs, however I couldn't get the world state display to consistently show up.

I removed world state related code from the ICC instance script and took to sending the world states manually (using `.debug worldstate` macros). Here, I found some fascinating results:
* If the first set of world states is sent _outside_ of a wing boss' boss room (`Putricide's Laboratory of Alchemical Horrors and Fun (4890)`, `Sanctum of Blood (4891)`, `The Frost Queen's Lair (4812)` and `The Frozen Throne (4859)`), the world state display will become stuck and _never_ show up, even if latter updates are sent in the correct area.
* If the first set of world states is sent _inside_ one of the aforementioned areas, the attempts tracker will properly show up and function even outside these areas - this includes being shown/hidden using the toggle outside these areas.

So yeah - it's a really weird client quirk, apparently, and we don't account for it.

tl;dr: Only send world states once the player has visited one of the boss rooms since relogging.

**Comment by ccrs** [comment]
icc has special worldstates that only show up on area change, iirc there is no other similar uses.

thing is there is no update sent on area change, so the states will never show. the reason why they do on load if the player is inside the area is well because the player is inside the area.

**Comment by Treeston** [comment]
Yeah, but if we send them before area change, they won't show up even if we send them again _after_ area change.

**Comment by Killyana** [comment]
@Keader close?

**Comment by Keader** [comment]
Hmm, searching yesterday, this issue is not valid.
When blizzard open Plagueworks, they put only 10 trys in PP. But after add Blood Queen, it go to 15 trys (PP + BQ) and after Sindragosa 20 trys...
I cant find when exactly it go to 50 trys, but i found that PP, BQ, Sindragosa and LK share 50 trys after all rooms was open in ICC, and after a while, blizzard removed it of normal instances, only having limited Attempts in Heroic.
